src/main/java/com/dataloom/datastore/edm/controllers/EdmController.java

```diff
@@ -84,6 +86,64 @@ public EntityDataModel getEntityDataModel() {
                 authorizedEntitySets::iterator );
     }
 
+    @Override
+    @RequestMapping(
+        method = RequestMethod.POST,
+        consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE )
+    @ResponseStatus( HttpStatus.OK )
+    public EdmDetails getEdmDetails( @RequestBody Set<EdmDetailsSelector> selectors) {
+        final Map<UUID, PropertyType> propertyTypes = new HashMap<>();
+        final Map<UUID, EntityType> entityTypes = new HashMap<>();
+        final Map<UUID, EntitySet> entitySets = new HashMap<>();
+        
+        selectors.forEach( selector -> {
```

Comment:
You can remove about 80% of this code if you expose getAll for each of the type maps in the EdmManager.